Rebuilt addtions for poi Info and verified.

IsJustaGhost opened the pull request:
I rebuilt poi matching and how the data is stored.
Now all pins for a zone are stored. This is for zones that exists on more than one zones. A specific parent can be chosen when using lib:GetZoneMapPinInfo.

Now pinInfo is acquired using lib:GetZoneMapPinInfo(zoneId, parentZoneId)
If the selected one has a map pin on it's parent, the pinInfo and, adjusted if needed, parentZoneId.
Below the function, I added how I am using this function and info as an example.

IsJustaGhost committed:
Update LibZone.lua

Added pin indices for zones that have pins on parent map.
Updated parentZoneIds to reflect geographical location.

This update is based off of the current version on esoui

IsJustaGhost committed:
Update LibZone.lua

Updated changes to match current pts.

IsJustaGhost committed:
Update LibZone.lua

Fixed missing predefined local poiDataTable

IsJustaGhost committed:
Update LibZone.lua

Added missing buildPoiDataTable(). Another thing I missed after changing from current release to pts.

IsJustaGhost committed:
Update LibZone.lua

Lastly, clear the poiDataTable after zoneData is created.

IsJustaGhost committed:
Update LibZone_Data.lua

Added lib.wayshrineString
Added lib.geographicalParentZoneIds

IsJustaGhost committed:
Update LibZone.lua

Moved lib.wayshrineString to LibZone_Data.lua
Moved lib.geographicalParentZoneIds to LibZone_Data.lua

IsJustaGhost committed:
Update LibZone.lua

Moved lib.wayshrineString to LibZone_Data.lua
Moved lib.geographicalParentZoneIds to LibZone_Data.lua

Restored parentZone
Added parentMapZoneId
- Since the purpose of this parent is to use with the map in one way or another

IsJustaGhost committed:
Update LibZone_Data.lua

Applied changes for improved poi matching.

IsJustaGhost committed:
Update LibZone.lua

Applied changes for improved poi matching.
Add function lib:GetZoneMapPinInfo

All maps in a list built with "for zoneIndex = 1, GetNumZones() do" matches map and poi, for zones with a map pin on parent, have been verified.

IsJustaGhost committed:
Update LibZone.lua

Removed debug text entries from data table.

Baertram commented:
Just saw that someone else got a problem with the parentIds and zones:
https://i.imgur.com/KpIUL1l.png

If you guys would maybe combine and test LibZones with his map addon this would be a good usecase imo?
I'm currently not able to work on this here as I'm very very busy and in vacations soon so please do some testing and improvements with your addons and usecase!

!MANY THANKS! @IsJustaGhost

IsJustaGhost committed:
Update LibZone.lua

Removed overlooked code from original poi change.

IsJustaGhost committed:
Update LibZone.lua

Added Faregrave as exception to set alternate parent as Faregrave instead of Fargrave City district

IsJustaGhost committed:
Update LibZone.lua

Add "parentZoneId = parentZoneId or GetParentZoneId(zoneId)" to lib:GetZoneMapPinInfo

IsJustaGhost committed:
Update LibZone.lua

Updated lib:GetZoneMapPinInfo

IsJustaGhost committed:
Update LibZone.lua

Renamed mapInfo to pinInfo

IsJustaGhost committed:
Update LibZone_Data.lua

added adjustedParentZoneIds

IsJustaGhost committed:
Update LibZone.lua

Added lib:GetZoneGeogrphicalParentZoneId

IsJustaGhost committed:
Update LibZone.lua

added lib:GetZoneGeogrphicalParentMapId

IsJustaGhost committed:
Update LibZone.lua

added lib:GetGeogrphicalParentMapId

IsJustaGhost committed:
Update LibZone.lua

updated lib:GetGeogrphicalParentMapId

IsJustaGhost committed:
Update LibZone.lua

Fixed typos

IsJustaGhost committed:
Update LibZone_Data.lua

Updated adjustedParentZoneIds
Added adjustedParentMultiZoneIds

IsJustaGhost committed:
Update LibZone.lua

Updated GetZonGeographicalParentZoneId
Ragnthar will first try to return the parent the player is in, or first entry.
The Harborage will use player alliance home.

IsJustaGhost committed:
Update LibZone.lua

Fixed zoneIndex, =

IsJustaGhost committed:
Update LibZone_Data.lua

IsJustaGhost committed:
Update LibZone.lua

IsJustaGhost committed:
Update LibZone.lua

IsJustaGhost committed:
Update LibZone.lua

Baertram commented:
I'll merge the pull request now and I'm working on a re-code of some variables, use of more locals and move of data to the data file.
I'll also add more description and details at the new api functions.
Please  wait for my update on that at the master branch and re-base your changes on MY changes then so your next pull request will include my changes, and your updated documentation.

THANK YOU!
